### DISCUSSION TITLE
Cumulative sum goal should set numerical habits to completed

### DISCUSSION THREAD

**Discussion by NomarCub** [iSoron/uhabits#1058] [Feature Requests]
I want to have a habit of running 20 kms a week. When I've run, say 8 kms 3 days in a row for a sum of 24, I want the `Hide completed` checkbox to hide the habit.

The days should only be within a rolling week, 3 consecutive being a simple example. In this case passing 20 within this timeframe should make the app consider the habit completed.

Right now setting the goal to 20 and the frequency to a week, setting the value to anything below the limit for even multiple days in a row doesn't complete the habit. The individual days don't add up, even though the habit is not daily, and they should be additive.

Related to https://github.com/iSoron/uhabits/discussions/949 and https://github.com/iSoron/uhabits/discussions/939.

**Comment by hiqua** [comment]
I think the behavior you describe makes sense, since if I want to track whether I've been able to run 20km once a week, I can just have a yes/no habit. Numerical habits are interesting if they sum towards a goal.

This probably requires a bit more thought though.

**Comment by ke352802081770314** [comment]
I was under the impression that the cumulative goal already works?
I have a goal "Gardening" with 150 minutes 1 times per week, and it shows a steady improvement in score, while I rarely hit 150 in a single day.
So maybe the issue you raise only refers to the "hide completed" feature and not the scoring.

**Reply by NomarCub** [reply]
Yes, I'm interested in `Hide completed` working. I said so in the first paragraph and meant that in the title.
